src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -440,15 +440,30 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, zinc_args,
       classpath_entry.directory_digest for classpath_entry in scalac_classpath_entries
     )
 
+    if self._zinc.use_native_image:
```

Comment:
I think it's pretty important to error here if there are `jvm_options` specified which we're going to ignore. Otherwise, I can easily imagine people wasting hours debugging why the options they're setting aren't having effect.

Reply:
1000% agreed on this

Reply:
The counterpoint is that _everything_ sets jvm_options, both upstream and downstream, because the `jvm_options` have default values. They also never contain logic.

Have said that: there is consensus. Will do it =)

Reply:
It's possible to explicitly set `jvm_options: []` whenever a native image is used for zinc to avoid this (which is why it might be better as an option on the zinc task instead of the subsystem, but since this is temporary anyway it should be fine).